src/core.ts

```diff
@@ -717,24 +709,28 @@ export class NodePart implements SinglePart {
     removeNodes(
         this.startNode.parentNode!, startNode.nextSibling!, this.endNode);
   }
+
+  commit() {
+    // nothing for now...
+  }
+
 }
 
 export type PartCallback =
     (instance: TemplateInstance, templatePart: TemplatePart, node: Node) =>
-        Part;
+        Part[];
 
 export const defaultPartCallback =
     (instance: TemplateInstance,
      templatePart: TemplatePart,
-     node: Node): Part => {
+     node: Node): Part[] => {
       if (templatePart.type === 'attribute') {
-        return new AttributePart(
-            instance, node as Element, templatePart.name!, templatePart.strings!
-        );
+        const comitter = new AttributeCommitter(node as Element, templatePart.name!, templatePart.strings!);
+        return comitter.parts;
       } else if (templatePart.type === 'node') {
-        return new NodePart(instance, node, node.nextSibling!);
+        return [new NodePart(instance, node, node.nextSibling!)];
       }
-      throw new Error(`Unknown part type ${templatePart.type}`);
+      throw [new Error(`Unknown part type ${templatePart.type}`)];
```

Comment:
Is this intentional?

Reply:
Yeah. I'll add some more comments, but now there's not a 1-1 correspondence between "template parts" and `Part`s. Maybe there's some further refinement there, but we need something with knowledge of a whole attribute.

Reply:
It just seems weird to me to be throwing an `Array<Error>`, especially because there's no special handling of this array anywhere in lit-html itself.

Reply:
oops, I didn't even see that it was an array of Error, I read this as a line returning multiple parts! Thanks!